File: Services/CategoryServices/DrivingSchoolApp.Services.CategoryServices.BCategoryAPI/DataAccerrss/Entities/Lesson.cs
﻿using DrivingSchoolApp.Services.CategoryServices.BCategoryAPI.DataAccerrss.Entities.Enums;

namespace DrivingSchoolApp.Services.CategoryServices.BCategoryAPI.DataAccerrss.Entities
{
	public class Lesson : Base
	{
		public LessonTypes Type { get; set; }
		public double Price { get; set; }
		public double TestPrice { get; set; }
		public string Description { get; set; }
		public string LicenceDescription { get; set; }
		public int Duration { get; set; }
		public double? VirtualPracticePrice { get; set; }
		public int? Discount { get; set; }
		public IEnumerable<CategoryLesson> CategoryLessons { get; set; } = new List<CategoryLesson>();
	}
}

File: Frontend/DrivingSchoolWeb/src/app/categories/category-d/models/lesson.ts
import {Base} from 'src/app/categories/shared/models/base';
import {LessonType} from 'src/app/categories/category-d/models/enums/lesson-type';

export interface Lesson extends Base{
  Type:LessonType,
  price: number,
  testPrice:number,
  description:string,
  licenceDescription: string,
  duration: number,
  virtualPracticePrice: number | null,
  discount:number | null
}
